recipes/microprofile/all/patches/sources_fix.patch

```diff
@@ -0,0 +1,60 @@
+--- a/microprofile.cpp
++++ b/microprofile.cpp
+@@ -4835,6 +4835,7 @@ void MicroProfileWriteSocket(void* Handle, size_t nSize, const char* pData)
+ }
+ 
+ #if MICROPROFILE_MINIZ
++#include <miniz/miniz.h>
+ #ifndef MICROPROFILE_COMPRESS_BUFFER_SIZE
+ #define MICROPROFILE_COMPRESS_BUFFER_SIZE (256<<10)
+ #endif
+@@ -8716,8 +8717,22 @@ int MicroProfileGetGpuTickReference(int64_t* pOutCPU, int64_t* pOutGpu)
+ 	return 1;
+ }
+ #elif MICROPROFILE_GPU_TIMERS_GL
++#include <glad/glad.h>
```

Comment:
Doesn't this fix the gl loader to glad?
Isn't microprofile gl loader independent?

Reply:
Yes, you are correct. But without it, microprofile would not even compile, as it doesn't include any OpenGL headers and uses OpenGL extension functions (such as `glGenQueries()`). Other viable solution I thought of was to add struct parameter to `MicroProfileGpuInitGL()` that would contain pointers to these OpenGL functions that microprofile uses. Since these functions would be provided by the consumer of this package, Glad dependency could be removed.

Reply:
I've removed the Glad dependency, but I had to patch the code so that the consumer of the package has to provide the pointers to the OpenGL functions that the upstream uses, if they want to use OpenGL timers.